File: godfather/roles/mafia/janitor.py
from godfather.roles.mixins import SingleAction, MafiaMember
from godfather.game.types import Priority

DESCRIPTION = 'You may clean a player at night.'


class Janitor(MafiaMember, SingleAction):
    name = 'Janitor'
    description = DESCRIPTION

    def __init__(self):
        super().__init__()
        self.action = 'clean'
        self.action_gerund = 'cleaning'
        self.action_priority = Priority.JANITOR
        self.action_text = 'clean a player'

    async def run_action(self, actions, player, target):
        record = actions.record[target.user.id]['nightkill']
        if record['result']:
            target.role.cleaned = True

    async def tear_down(self, actions, player, target):
        record = actions.record[target.user.id]['nightkill']
        if record['result']:
            await player.user.send('You secretly know that your target\'s role was {}.'.format(target.role.name))

File: src/lib/mafia/roles/mafia/Janitor.ts
import MafiaRole from '@mafia/mixins/MafiaRole';
import type Player from '@mafia/structures/Player';
import { CleanAction } from '../../actions/common/CleanAction';
import type { NightAction } from '../../managers/NightAction';
import { ActionRole } from '../../structures/ActionRole';

class Janitor extends ActionRole {
	public name = 'Janitor';
	public actions: NightAction[] = [new CleanAction(this)];

	public constructor(player: Player) {
		super(player);
		this.description = this.game.t('roles/neutral:janitorDescription');
	}
}

Janitor.categories = [...Janitor.categories, 'Mafia Deception'];

export default MafiaRole(Janitor);
